### PR TITLE
Added node and type selector UI to Error Logs screen

### PR DESCRIPTION
**Pull request by jinapurapu**
**What it does:** Adds UI to allow node, user Agent and log type selection for Error logs. It can take several minutes after clicking Start to begin picking up logs with a node selected, please be patient.

**What it looks like:**Screenshot from 2022-04-01 20-25-55
Screenshot from 2022-04-01 20-26-04
Screenshot from 2022-04-01 20-26-20
Screenshot from 2022-04-01 20-23-40

Selecting different nodes gives different results:
Screenshot from 2022-04-01 19-05-36

 
Addresses https://github.com/miniohq/engineering/issues/508

### CODE REVIEW

File: restapi/configure_console.go [hunk 1]
```diff
@@ -128,6 +128,8 @@ func configureAPI(api *operations.ConsoleAPI) http.Handler {
 	registerAdminTiersHandlers(api)
 	//Register Inspect Handler
 	registerInspectHandler(api)
+	// Register nodes handlers
+	registersNodesHandler(api)
```
Comment (kaankabalak) [inline restapi/configure_console.go:132]: Very minor issue here, should we change the function name to `registerNodesHandler` to keep it consistent with the other functions?

```suggestion
	registerNodesHandler(api)
```

File: restapi/admin_nodes.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"context"
+	//"sort"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/minio/console/models"
+	"github.com/minio/console/restapi/operations"
+	"github.com/minio/console/restapi/operations/admin_api"
+)
+
+func registersNodesHandler(api *operations.ConsoleAPI) {
```
Comment (kaankabalak) [inline restapi/admin_nodes.go:29]: Similar to my previous comment, should we change the name of the function here to `registerNodesHandler` to keep things consistent?

```suggestion
func registerNodesHandler(api *operations.ConsoleAPI) {
```

File: portal-ui/src/screens/Console/Logs/ErrorLogs/ErrorLogs.tsx [hunk 3]
```diff
@@ -170,38 +231,96 @@ const ErrorLogs = ({
...
+              <CheckboxWrapper
+                checked={logType === "minio" || allTypes}
+                id={"minio_type"}
+                name={"minio_type"}
+                label={"Minio"}
+                onChange={(item) => {
+                  setAllTypes(false);
+                  item.target.checked ? setLogType("minio") : setLogType("");
+                }}
+                value={"Minio"}
```
Comment (kaankabalak) [inline portal-ui/src/screens/Console/Logs/ErrorLogs/ErrorLogs.tsx:303]: Another minor issue, should we change the `label` and `value` here from `Minio` to `MinIO`?

```suggestion
                label={"MinIO"}
                onChange={(item) => {
                  setAllTypes(false);
                  item.target.checked ? setLogType("minio") : setLogType("");
                }}
                value={"MinIO"}
```

**Comment by prakashsvmx** [comment]
@jinapurapu  could you please resolve the conflicts ?

**Review by dvaldivia — DISMISSED** [review]

**Review by bexsoft — CHANGES_REQUESTED** [review]
No Information is retrieved with default settings after clicking on the Start Logs button. I think it would be better to change the Log type default value to "All".
Screen Shot 2022-03-31 at 12 42 50
Screen Shot 2022-03-31 at 12 42 43



Also it would be nice to have the selectors in full width alignment and put the Filter input bellow those fields as the action of that field has to work with the results

Screen Shot 2022-03-31 at 12 41 20
Screen Shot 2022-03-31 at 12 41 25


Beside of that, Works good!

File: restapi/admin_console.go [hunk 4]
```diff
@@ -29,14 +29,14 @@ import (
...
+func startConsoleLog(ctx context.Context, conn WSConn, client MinioAdmin, logRequest LogRequest) error {
+
 	// name of node, default = "" (all)
-	node := ""
+	node := logRequest.node
 	// number of log lines
 	lineCount := 100
 	// type of logs "minio"|"application"|"all" default = "all"
-	logKind := "all"
+	logKind := logRequest.logType
```
Comment (Alevsk) [inline restapi/admin_console.go:39]: Please validate the provided value is one of those 3 options: minio, application or all, default to all if an invalid value is provided

Comment (Alevsk) [inline restapi/admin_console.go:39]: Please validate the provided value is one of those 3 options: `minio`, `application` or `all`, default to `all` if an invalid value is provided

**Review by Alevsk — CHANGES_REQUESTED** [review]

**Comment by Alevsk** [comment]
image

This is more like a visual bug but theres a css class making the first letter of every word uppercase on the list of nodes, @bexsoft this is the default behavior for that component?

**Review by Alevsk — CHANGES_REQUESTED** [review]

**Comment by bexsoft** [comment]
With latest changes I'm not able to retrieve any information, are there any special steps to follow?

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]
LGTM

**Pull request merged**